### ISSUE TITLE
The DataGridViewComboBoxColumn cannot be displayed as combobox mode when the DisplayStyle property is set to Nothing.

### ISSUE THREAD

**Issue by Jessie-Zhang01** [dotnet/winforms#1587]
* .NET Core Version: (.Net Core SDK 5.0.100.13662 build from master branch)
* Have you experienced this same bug with .NET Framework?: Yes

**Steps to reproduce:**

1. Create a Winforms .Net Core application.
2. Add DataGridView control, then add a DataGridViewComboBoxColumn column.
3. Set the DisplayStyle property of DataGridViewComboBoxColumn as Nothing.
4. Build and run it.
[WindowsFormsApp6.zip](https://github.com/dotnet/winforms/files/3479985/WindowsFormsApp6.zip)

**Actual:** 
The DataGridViewComboBoxColumn cannot be displayed as combobox mode when the DisplayStyle property is set to Nothing.

**Expected:** 
From Merrie comment：Please file a GitHub issue to add the Obsolete attribute for the type so that we can warn developers not to use it due to accessibility issues.

**Comment by r-aghaei** [comment]
@Jessie-Zhang01  When the cell is not in edit mode, it doesn't display a combo box, but when the cell is in edit mode, it shows the combo box as expected.

It's already mentioned in the [documentations](https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.datagridviewcomboboxcolumn.displaystyle?view=netframework-4.8#remarks):

> The `DisplayStyle` property affects the appearance of cells in the column when they are not in edit mode, and regardless of whether they are read-only. When a cell is in edit mode, it always looks like a `ComboBox` control.

The code which you shared shows expected behavior based on the documentations.

**Comment by Jessie-Zhang01** [comment]
Yes, the current state of the product is like this, but @merriemcgaw wants to add some accessibility improvement in future, so file this Github issue to track it.

**Comment by r-aghaei** [comment]
IMO it's a feature and works as expected. I believe even if someone decides to create new column types like `DatePickerColumn`, `NumericUpDownColumn`, `ColorPickerColumn`, ... they should follow similar practice to allow developers choose rendering mode of the cell when the cell is not in edit mode.

**Comment by RussKie** [comment]
Whilst the feature is working as expected it is an accessibility issue, and that is what we'll need to address by disallowing use of `Nothing` state.

**Comment by weltkante** [comment]
I don't think disallowing distinct display/editor states is the way to go, every other grid also has this behavior of only showing the editor on the edited cell and it is very much desired by end users. The most common usage of a data grid is to view data, with an occasional desired to edit, not edit every single cell, forcing editing UI on the user is very distracting. Removing this capability will just lead to people abandoning the builtin grid because of poor functionality.

If anything you need to improve your accessibility mapping to properly report cells not in edit state in the desired way, not disable widely used features.

**Comment by r-aghaei** [comment]
The users can easily start editing and see the editing control by pressing <kbd>F2</kbd> or any key by default, or even as soon as they enter the cell (if `EditMode` set to `EditOnEnter`).

Consider the following examples:
* Label editing in `TreeView` or `ListView` all over Windows. 
* `PropertyGrid` control doesn't show editing control, unless you select the property.

In fact, for a control like `DataGridView`, I believe it should be defined as an standard behavior for rest of possible column types as well. (Like `DatePickerColumn` and so on. The control should be able to render the cell without editing control, unless the cell starts editing.)

**Comment by merriemcgaw** [comment]
Thank you all for the feedback, it's much appreciated. I'll reach out to our accessibility team and find out what their expected behavior is in this case. The tooling flagged a mismatch between the reported type of AccessibleObject and the visible one. I'm hopeful we can find a way to alleviate their concerns and satisfy our customers.

**Comment by Lydia-Shi** [comment]
Verified this issue with .NET 7.0.100-alpha.1.21566.13, this issue is fixed.
![image](https://user-images.githubusercontent.com/54019172/142154152-8623fbaf-5949-452b-8ba6-07c0fcf741be.png)
